jsr352/core/src/main/java/org/hibernate/search/jsr352/logging/impl/Log.java

```diff
@@ -0,0 +1,227 @@
+	@Message(id = JSR_352_MESSAGES_START_ID + 11,
+			value = "Session is closed."
+	)
+	void sessionClosed();
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 12,
+			value = "Stateless session is closed."
+	)
+	void statelessSessionClosed();
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 13,
+			value = "Scrollable results are closed."
+	)
+	void scrollableResultsClosed();
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 14,
+			value = "Entity manager is closed."
+	)
+	void entityManagerClosed();
+
+	@LogMessage(level = ERROR)
+	@Message(id = JSR_352_MESSAGES_START_ID + 15,
+			value = "Unable to close session."
+	)
+	void unableToCloseSession(@Cause Exception e);
+
+	@LogMessage(level = ERROR)
+	@Message(id = JSR_352_MESSAGES_START_ID + 16,
+			value = "Unable to close stateless session."
+	)
+	void unableToCloseStatelessSession(@Cause Exception e);
+
+	@LogMessage(level = ERROR)
+	@Message(id = JSR_352_MESSAGES_START_ID + 17,
+			value = "Unable to close scrollable results."
+	)
+	void unableToCloseScrollableResults(@Cause Exception e);
+
+	@LogMessage(level = ERROR)
+	@Message(id = JSR_352_MESSAGES_START_ID + 18,
+			value = "Unable to close entity manager."
+	)
+	void unableToCloseEntityManager(@Cause Exception e);
+
+	@LogMessage(level = INFO)
+	@Message(id = JSR_352_MESSAGES_START_ID + 19,
+			value = "Optimizing all entities ..."
+	)
+	void startOptimization();
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 20,
+			value = "Closing everything..."
+	)
+	void closingEverything();
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 21,
+			value = "%1$d criteria found."
+	)
+	void criteriaSize(int size);
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 22,
+			value = "Checkpoint reached. Sending checkpoint ID to batch runtime... (entity='%1$s', id='%2$s')"
+	)
+	void checkpointReached(String entityName, Object checkpointId);
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 23,
+			value = "[partitionId=%1$d] Prepare reader for entity '%2$s'."
+	)
+	void prepareReaderFor(int partitionId, String entityName);
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 24,
+			value = "%1$s"
+	)
+	void printBound(PartitionBound partitionBound);
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 25,
+			value = "Reading entities..."
+	)
+	void readingEntities();
+
+	@LogMessage(level = INFO)
+	@Message(id = JSR_352_MESSAGES_START_ID + 26,
+			value = "No more results, read ends."
+	)
+	void noMoreResults();
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 27,
+			value = "Processing entity with id: '%1$s'"
+	)
+	void processEntity(Object entityId);
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 28,
+			value = "%1$s is closed.")
+	void isClosed(String simpleClassName);
```

Comment:
Actually, I was planned to use this method for logging the method `close()` in both `EntityReader` and `LuceneDocWriter`, but I only added it to the writer... That's why the method looks so weird. The `simpleClassName` should be either `EntityReader` or `LuceneDocWriter`.

I'll rework it.